bundles/automation/org.eclipse.smarthome.automation.provider.file/src/main/java/org/eclipse/smarthome/automation/internal/provider/file/AbstractFileProvider.java

```diff
@@ -256,9 +256,19 @@ protected void importFile(String parserType, URL url) {
 
     protected void updateProvidedObjectsHolder(URL url, Set<E> providedObjects) {
         if (providedObjects != null && !providedObjects.isEmpty()) {
+            List<String> previousPortfolio = providerPortfolio.get(url);
             List<String> uids = new ArrayList<String>();
             for (E providedObject : providedObjects) {
                 String uid = getUID(providedObject);
+                if (providedObjectsHolder.get(uid) == null) {
+                    if (checkExistence(uid)) {
+                        continue;
+                    }
+                } else if (previousPortfolio == null || !previousPortfolio.contains(uid)) {
+                    logger.error("{} with UID \"{}\" already exists! Failed to create a second with the same UID!",
```

Comment:
In what kind of situation can we get here? If we have multiple files that contain entities with the same uid?
"error" logging should only be used in irrecoverable situations - I would think that "warning" should be enough here to tell the user that he has some problematic files and thus the second entity is simply ignored.